### PR TITLE
Testcase for indexof on strings

### PR DESCRIPTION
**Pull request by ralfhandl**

### CODE REVIEW

File: abnf/odata-abnf-testcases.yaml [hunk 1]
```diff
@@ -1811,6 +1811,10 @@ TestCases:
     Rule: boolCommonExpr
     Input: endswith(["Fred","George","Ron"],["George","Ron"])
 
+  - Name: 5.1.1.5.3 indexof
+    Rule: boolCommonExpr
```
Comment (HeikoTheissen) [inline abnf/odata-abnf-testcases.yaml:1815]:
```suggestion
    Rule: commonExpr
```

**Review by HeikoTheissen — CHANGES_REQUESTED** [review]
`indexof` does not give a boolean

File: abnf/odata-abnf-testcases.yaml (hunk 1 above)
Comment (ralfhandl) [reply abnf/odata-abnf-testcases.yaml]: Corrected also in a few other places

File: abnf/odata-abnf-testcases.yaml [hunk 2]
```diff
@@ -1812,19 +1812,19 @@ TestCases:
     Input: endswith(["Fred","George","Ron"],["George","Ron"])
 
   - Name: 5.1.1.5.3 indexof
-    Rule: boolCommonExpr
+    Rule: commonExpr
     Input: indexof(CompanyName,'lfreds')
 
   - Name: 5.1.1.5.4 indexof
-    Rule: boolCommonExpr
+    Rule: commonExpr
```
Comment (HeikoTheissen) [inline abnf/odata-abnf-testcases.yaml:1819]: But this and the next two *are* `boolCommonExpr`.

Comment (ralfhandl) [reply abnf/odata-abnf-testcases.yaml:1819]: indexof returns a number

File: abnf/odata-abnf-testcases.yaml [hunk 3]
```diff
@@ -1811,16 +1811,20 @@ TestCases:
...
+    Input: indexof(CompanyName,'lfreds')
+
   - Name: 5.1.1.5.4 indexof
-    Rule: boolCommonExpr
+    Rule: commonExpr
     Input: indexof(["Fred","George","Ron"],["George","Ron"]) eq 1
 
   - Name: 5.1.1.5.5 length
-    Rule: boolCommonExpr
+    Rule: commonExpr
```
Comment (HeikoTheissen) [inline abnf/odata-abnf-testcases.yaml:1823]:
```suggestion
    Rule: boolCommonExpr
```
The whole expression returns a boolean.

File: abnf/odata-abnf-testcases.yaml [hunk 4]
```diff
@@ -1811,16 +1811,20 @@ TestCases:
...
     Input: length(CompanyName) eq 19
 
   - Name: 5.1.1.5.5 length
-    Rule: boolCommonExpr
+    Rule: commonExpr
```
Comment (HeikoTheissen) [inline abnf/odata-abnf-testcases.yaml:1827]:
```suggestion
    Rule: boolCommonExpr
```
The whole expression returns a boolean.

**Review by HeikoTheissen — CHANGES_REQUESTED** [review]

**Review by HeikoTheissen — APPROVED** [review]

**Pull request merged**